homeassistant/components/heos/config_flow.py

```diff
@@ -0,0 +1,25 @@
+"""Config flow to configure Heos."""
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+
+from .const import DOMAIN
+
+
+def format_title(host: str) -> str:
+    """Format the title for config entries."""
+    return "Controller ({})".format(host)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class HeosFlowHandler(config_entries.ConfigFlow):
+    """Define a flow for HEOS."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_import(self, user_input=None):
```

Comment:
step_user is missing.

Reply:
This does not add UI-initiated setup, only from config.

Reply:
So why have config entries? The whole point is to enable configuration from the frontend.

Reply:
That's not true, but regardless, as mentioned in the PR description, that will be added in a new PR because that does add quite a bit more code and tests.  This PR lays the foundation.